docs/source/tutorials/loading_pretrained_lenses.rst

```diff
@@ -12,35 +12,39 @@ Loading a pre-trained lens
 First check if there is a pre-trained lens available in our spaces' `pre-trained lenses folder`_.
 
 Once you have found a lens that you want to use, you can load it using the simply load it
-and its corresponding tokenizer using the hugging face API.
+and its corresponding tokenizer using the hugging face API. A tuned lens is always associated with
+a model that was used to train it so first load the model and then the lens.
 
 >>> import torch
 >>> from tuned_lens import TunedLens
 >>> from transformers import AutoModelForCausalLM
 >>> device = torch.device('cpu')
 >>> model = AutoModelForCausalLM.from_pretrained('EleutherAI/pythia-160m-deduped')
->>> tuned_lens = TunedLens.load("pythia-160m-deduped", map_location=device)
+>>> tuned_lens = TunedLens.from_model_and_pretrained(model, "pythia-160m-deduped", map_location=device)
```

Comment:
Oh `from_model_and_pretrained` seems too long to me, could we just do `from_pretrained` or `from_model`?

HuggingFace models have a `name_or_path` attribute which stores the HF Hub model string that you would use to download it. A `from_model` method could use this attribute to try to load the appropriate tuned lens weights.

Conversely, `TunedLens.from_pretrained` could call `AutoModelForCausalLM.from_pretrained` internally in order to get the unembedding. Or we could store the unembedding in the tuned lens checkpoint itself, as we currently do, in order to keep the lens checkpoint self-contained.

Reply:
That would be super dicey if you ever want to distribute Llama lenses.

Reply:
I've heard that LAION is distributing _diff_ weights for LLaMA to avoid legal troubles, the idea being that the diff weights are ~useless without the original checkpoint. I think we could make a similar case here

Reply:
But idk, I don't have a strong opinion on this. There are certain cases where you like, have saved hidden states and it's nice to be able to directly apply the tuned lens to them without also loading the model. But they are kinda niche

Reply:
I'd be happy to add a wrapper that automatically pulls down the model as `from_pretrained`. It's just a bit annoying since it will load the model into memory twice which could cause problems for massive models like `EleutherAI/gpt-neox-20b`. From model restricts us to having a one-to-one relationship between lenses and models in the future which may not always make sense. The main reason I stopped storing the undemanding matrix as part of the lens was to separate concerns between the `Unembed` and the `TunedLens` components. I wanted to keep the amount of stuff we actually save to disk to a minimum to make future code changes easier. All of this is to say that's why a settled on the `from_model_and_pretrained` interface.

Another option could be something like this `model, lens = TunedLens.from_pretrained('EleutherAI/gpt-neox-20b')` but idk that feels a bit weird.